code/game/objects/structures/fountain.dm

```diff
@@ -59,10 +59,23 @@
 	desc = "The water flows beautifully from the spout, but the water in the pool does not ripple."
 
 /obj/structure/fountain/mundane
-	name = "fountain"
-	desc = "A beautifully constructed fountain."
-	icon_state = "fountain_g"
-	used = TRUE
+	name                   = "fountain"
+	desc                   = "A beautifully constructed fountain."
+	icon_state             = "fountain_g"
+	tool_interaction_flags = TOOL_INTERACTION_DECONSTRUCT
+	w_class                = ITEM_SIZE_STRUCTURE
+	material               = /decl/material/solid/stone/marble 
+	used                   = TRUE
+
+/obj/structure/fountain/mundane/Initialize(ml, _mat, _reinf_mat)
+	. = ..()
+	atom_flags |= ATOM_FLAG_OPEN_CONTAINER | ATOM_FLAG_CLIMBABLE
+	setup_reagents(ml)
+
+/obj/structure/fountain/mundane/proc/setup_reagents(var/should_fill = TRUE)
```

Comment:
Why is this a separate proc rather than just `if(ml)` in Initialize()?

Reply:
Mainly because its easier to override this way than just piling up more things directly into initialize. And also it makes my life easier downstream.